sdk/avalon_sdk/fabric/fabric_work_order_receipt.py

```diff
@@ -0,0 +1,331 @@
+                return -1
+            if not is_valid_hex_str(
+                    binascii.hexlify(updater_id).decode("utf8")):
+                logging.info("Invalid updater id {}".format(updater_id))
+                return -1
+            params = []
+            params.append(work_order_id)
+            params.append(updater_id)
+            params.append(update_type)
+            params.append(update_data)
+            if update_signature is not None:
+                params.append(update_signature)
+                params.append(signature_rules)
+            work_order_result = self.__fabric_wrapper.invoke_chaincode(
+                self.CHAIN_CODE,
+                'workOrderReceiptUpdate',
+                params)
+            if work_order_result is not None:
+                return work_order_result
+            else:
+                return -1
+        else:
+            logging.error(
+                "Fabric wrapper instance is not initialized")
+            return -1
+
+    def work_order_receipt_retrieve(self, work_order_id):
+        """
+        Retrieving a Work Order Receipt
+        Inputs:
+        work_order_id is the id of the Work Order to be retrieved.
+        id is used for json rpc request
+        Outputs:
+        worker_service_id, requester_id, work_order_id, receipt_create_status,
+        and work_order_request_hash are defined in work_order_receipt_create().
+        None on error
+        """
+        if (self.__fabric_wrapper is not None):
+            if not is_valid_hex_str(
+                    binascii.hexlify(work_order_id).decode("utf8")):
+                logging.info("Invalid work order id {}".format(work_order_id))
+                return -1
+            params = []
+            params.append(work_order_id)
+            retrieve_result = self.__fabric_wrapper.invoke_chaincode(
+                self.CHAIN_CODE,
+                'workOrderReceiptRetrieve',
+                params)
+            if retrieve_result is not None:
+                return retrieve_result
+            else:
+                return -1
+        else:
+            logging.error(
+                "Fabric wrapper instance is not initialized")
+            return -1
+
+    def work_order_receipt_update_retrieve(self, work_order_id,
+                                           updater_id, update_index):
+        """
+        Function to retrieve update to receipt
+        Params
+            work_order_id is a Work Order id that was
+            sent in the corresponding work_order_submit request.
+            updater_id is an id of the updating entity. It is optional
+            update_index is an index of the update to retrieve.
+            Value "0xFFFFFFFF" is reserved to retrieve the last
+            received update.
+        Returns
+            updater_id, update_type, update_data, update_signature, and
+            signature_rules are defined work_order_receipt_update().
```

Comment:
The function is returning -1 or error and work_order_result otherwise. looks like inconsistent return value.

Reply:
Changed